### PR TITLE
[CopyFilesV2] Considered delay between retries for cp action. Switched to node 6.

### PR DESCRIPTION
**Pull request by anatolybolshakov**
**Task name**: CopyFilesV2

**Description**: we need to consider delayBetweenRetries input for cp action also - to make retries more effective agains flakiness issues.
Also reworked retry logic in retryHelper:
* removed extra methods for retry;
* fixed message that shows the number of remaining attempts;
* fixed undefined instead of a function name in the error message;
* added option to ignore ENOENT for tl.stats function

**Documentation changes required:**N

**Added unit tests:** N

**Attached related issue:** N

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it - bumped only patch since we are going to hotfix.
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/CopyFilesV2/copyfiles.ts [hunk 1]
```diff
@@ -179,7 +179,9 @@ async function main(): Promise<void> {
                         console.log(tl.loc('FileAlreadyExistAt', file, targetPath));
                     } else { // copy
                         console.log(tl.loc('CopyingTo', file, targetPath));
-                        tl.cp(file, targetPath, undefined, undefined, retryCount);
+                        await retryHelper.RunWithRetry(
+                            () => tl.cp(file, targetPath, undefined, undefined)
```
Comment (DaniilShmelev) [inline Tasks/CopyFilesV2/copyfiles.ts:183]:
```suggestion
                            () => tl.cp(file, targetPath)
```

File: Tasks/CopyFilesV2/copyfiles.ts [hunk 2]
```diff
@@ -218,8 +220,10 @@ async function main(): Promise<void> {
                             () => fs.chmodSync(targetPath, targetStats.mode | 146),
                             targetPath);
                     }
+                    await retryHelper.RunWithRetry(
+                        () => tl.cp(file, targetPath, "-f", undefined)
```
Comment (DaniilShmelev) [inline Tasks/CopyFilesV2/copyfiles.ts:224]:
```suggestion
                        () => tl.cp(file, targetPath, "-f")
```

**Review by DaniilShmelev — APPROVED** [review]
Take a look at the comments, other than that LGTM

File: Tasks/CopyFilesV2/copyfiles.ts [hunk 3]
```diff
@@ -37,6 +37,23 @@ function makeDirP(targetFolder: string, ignoreErrors: boolean): void {
...
+ * If ignoreEnoent is false ENOENT will be thrown from the function.
+ * @param path path for which methid will try to get tl.FsStats.
+ * @param ignoreEnoent ignore ENOENT error during check of path stats.
+ * @returns 
+ */
+function stats(path: string, ignoreEnoent: boolean): tl.FsStats {
+    try {
+        return tl.stats(path);
+    } catch (err) {
+        if (err.code != 'ENOENT' && ignoreEnoent) {
```
Comment (anatolybolshakov) [inline Tasks/CopyFilesV2/copyfiles.ts:51]: If ignoreEnoent is false - it won't throw exception, although it should

File: Tasks/CopyFilesV2/copyfiles.ts [hunk 4]
```diff
@@ -150,23 +164,20 @@ async function main(): Promise<void> {
 
                 if (!createdFolders[targetDir]) {
                     await retryHelper.RunWithRetry(
-                        () => makeDirP(targetDir, ignoreMakeDirErrors));
+                        () => makeDirP(targetDir, ignoreMakeDirErrors),
+                        `makeDirP for ${targetDir}`
```
Comment (anatolybolshakov) [inline Tasks/CopyFilesV2/copyfiles.ts:168]: Could it make sense to replace it by human-readable description of action? 😃

File: Tasks/CopyFilesV2/copyfiles.ts [hunk 5]
```diff
@@ -37,6 +37,23 @@ function makeDirP(targetFolder: string, ignoreErrors: boolean): void {
     }
 }
 
+/**
+ * Gets stats for the provided path. Will ignore ENOENT error if ignoreEnoent is true.
+ * If ignoreEnoent is false ENOENT will be thrown from the function.
+ * @param path path for which methid will try to get tl.FsStats.
```
Comment (DaniilShmelev) [inline Tasks/CopyFilesV2/copyfiles.ts:43]:
```suggestion
 * @param path path for which method will try to get tl.FsStats.
```

**Comment by anatolybolshakov** [comment]
Tested changes locally, looks good!

**Comment by anatolybolshakov** [comment]
LGTM! (can't approve since I created it originally)

**Review by EzzhevNikita — APPROVED** [review]

**Pull request merged**